### PR TITLE
Server: Add client connection timeout

### LINKED ISSUE #1: Hangs after a number of client disconnects
When I try to connect multiple clients then disconnect them before completing their transfer, after a while the server stops receiving new clients.

### PR DESCRIPTION
**Pull request by Abogical**
Fixes #1

### CODE REVIEW

File: src/tftp/server.rs [hunk 1]
```diff
@@ -95,24 +96,30 @@ fn handle_client(socket: UdpSocket, mut server: TFTPServer, client_addr: SocketA
...
+            }
+        }
     }
     // });
 }
 
 pub fn handle_new_client(client_addr: SocketAddr, rq_packet: &[u8]) {
     println!("New connection: {}", client_addr);
     let socket = UdpSocket::bind("0.0.0.0:0").expect("Failed to bind UDP socket");
+    socket.set_read_timeout(Some(Duration::from_secs(5)));
```
Comment (shakram02) [inline src/tftp/server.rs]: Can we move the timeout `Duration` to a constant?

File: src/tftp/server.rs [hunk 2]
```diff
@@ -95,24 +96,30 @@ fn handle_client(socket: UdpSocket, mut server: TFTPServer, client_addr: SocketA
...
-        server.run(raw_msg);
+                server.run(raw_msg);
+            },
+            Err(e) => {
+                eprintln!("Client connection error: {}", e);
+                break;
+            }
+        }
     }
     // });
```
Comment (shakram02) [inline src/tftp/server.rs]: please remove this also, I forgot to remove it

**Review by shakram02 — CHANGES_REQUESTED** [review]

**Comment by Abogical** [comment]
Done!

File: src/tftp/server.rs [hunk 3]
```diff
@@ -95,24 +96,30 @@ fn handle_client(socket: UdpSocket, mut server: TFTPServer, client_addr: SocketA
...
+                eprintln!("Client connection error: {}", e);
+                break;
+            }
+        }
     }
-    // });
 }
 
 pub fn handle_new_client(client_addr: SocketAddr, rq_packet: &[u8]) {
+    const sock_dur: Option<Duration> = Some(Duration::from_secs(5));
```
Comment (shakram02) [inline src/tftp/server.rs]: `sock_dur` isn't a function constant (as it's a program property), so it's best placed at the top of the file

**Review by shakram02 — APPROVED** [review]

**Pull request merged**